ext/extconf.rb

```diff
@@ -7,8 +7,55 @@
   $CPPFLAGS += $CXXFLAGS
 end
 
+brew = find_executable('brew')
+
+# Automatically use homebrew to discover thrift package if it is
+# available and not disabled via --disable-homebrew.
+use_homebrew = enable_config('homebrew', brew)
+
+if use_homebrew
+  homebrew_package = with_config('homebrew-thrift-package', 'thrift@0.9')
+  $stderr.puts "using Homebrew to locate thrift package '#{homebrew_package}'"
+
+  thrift_prefix = `#{brew} --prefix #{homebrew_package}`.strip
+
+  unless File.exists?(thrift_prefix)
+    $stderr.puts "#{thrift_prefix} does not exist"
```

Comment:
do we want to include something like `or pass --disable-homebrew if you installed thrift outside of homebrew` so they can see directly in the error message that this is another mechanism for recovery?